src/shared/condition.c

```diff
@@ -265,6 +267,61 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_osrelease(Condition *c, char **env) {
+        const char *parameter = c->parameter;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_OS_RELEASE);
+
+        for (;;) {
+                _cleanup_free_ char *key = NULL, *condition = NULL, *actual_value = NULL;
+                OrderOperator order;
+                const char *word;
+                bool matches;
+
+                r = extract_first_word(&parameter, &condition, NULL, EXTRACT_UNQUOTE);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse parameter: %m");
+                if (r == 0)
+                        break;
+
+                /* parse_order() needs the string to start with the comparators */
+                word = condition;
+                r = extract_first_word(&word, &key, "!<=>", EXTRACT_RETAIN_SEPARATORS);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse parameter: %m");
+                /* The os-release spec mandates env-var-like key names */
+                if (r < 1 || isempty(word) || !env_name_is_valid(key))
```

Comment:
i think "r == 0" rather than "r < 1" would be more readable.

Reply:
i'd drop the isempty(word) check here. it appears redundant, the parse_order() below should catch that, no?

Reply:
> i'd drop the isempty(word) check here. it appears redundant, the parse_order() below should catch that, no?

'word' can be NULL, which means parse_order will assert (via `startswith()`). Rather than just checking for non-null, might as well check that it's not empty either since we got the helper.

Reply:
changed to `r == 0`